frontend/src/app/app.tsx

```diff
@@ -10,12 +13,15 @@ import {
     useAppSelector,
     useEffect,
 } from '~/bundles/common/hooks/hooks.js';
+import { type RootReducer } from '~/framework/store/store.js';
+
+const getAuthState = (state: RootReducer): State => state.auth;
 
 const App: React.FC = () => {
     const dispatch = useAppDispatch();
-    const { dataStatus } = useAppSelector(({ auth }) => ({
-        dataStatus: auth.dataStatus,
-    }));
+    const dataStatus = useAppSelector(
+        (rootState) => getAuthState(rootState).dataStatus,
+    );
```

Comment:
Why we need this changes ?

Reply:
@MaksymGavatuk previous code structure caused a great number of console warnings
`useSelector.js:63 Selector unknown returned a different result when called with the same parameters. This can lead to unnecessary rerenders.
Selectors that return a new reference (such as an object or an array) should be memoized`